lightning/src/onion_message/blinded_path.rs

```diff
@@ -58,14 +58,30 @@ pub struct BlindedHop {
 }
 
 impl BlindedPath {
+	/// Create a blinded path to be forwarded along `node_pks`. The last node pubkey in `node_pks`
+	/// will be the destination node and will receive the `path_id`.
+	///
+	/// Errors if less than two hops are provided or if `node_pk`(s) are invalid.
+	pub fn with_id<ES: Deref, T: secp256k1::Signing + secp256k1::Verification>(
```

Comment:
Hmm, I don't really see the point of a separate method vs an `Option`al path_id, it kind of blows up the number of methods after the payment path methods are added. 

Also, will this method be used in this PR at some point? I can't tell if it's orthogonal to the PR

Reply:
> Hmm, I don't really see the point of a separate method vs an `Option`al path_id, it kind of blows up the number of methods after the payment path methods are added.
> 

Eh, I'm not a fan of having to pass `None`, especially when the number of parameters starts to grow. Isn't very readable at the the call sites. Might as well write a builder at that point.

> Also, will this method be used in this PR at some point? I can't tell if it's orthogonal to the PR

Currently a work in process. It's for adding the a payment secret to the blinded payment path. So it seems we'll only have two methods in the end regardless?